ocaml/idl/latex_backend.ml

```diff
@@ -18,92 +18,52 @@ open Datamodel
-      h :: t -> (h, if t = [] then "" else List.hd t)
-    | [] -> assert false
-  in
-  let prespace, postspace = split_in_2 ' ' s in
-  let preeol, posteol = split_in_2 '\n' s in
-
-  if String.length prespace < String.length preeol then
-    (Format.fprintf formatter "%s@ " prespace;
-     if String.length postspace > 0 then
-       formatted_wrap formatter postspace)
-  else
-    (if String.length posteol > 0 then
-       (Format.fprintf formatter "%s@\n" preeol;
-        formatted_wrap formatter posteol)
-     else
-       Format.fprintf formatter "%s@ " preeol)
-
-
-let wrap s =
-  let buf = Buffer.create 16 in
-  let formatter = Format.formatter_of_buffer buf in
-
-  Format.pp_open_hvbox formatter 0;
-  Format.pp_set_margin formatter 76;
-  formatted_wrap formatter s;
-  Format.pp_close_box formatter ();
-
-  Format.fprintf formatter "%!";
-
-  Buffer.contents buf
+open Stdext
+open Xstringext
+open Pervasiveext
 
 let escape s =
   let sl = String.explode s in
-  let in_quote = ref true in
   let esc_char =
     function
-      '"' -> in_quote := not !in_quote; if !in_quote then "''" else "``"
     | '_' -> "\\_"
     | '#' -> "\\#"
     | c -> String.make 1 c in
   let escaped_list = List.map esc_char sl in
   String.concat "" escaped_list
 
-let full_stop s =
-  if String.length s = 0 then s^"."
-  else
-  if String.get s (String.length s - 1) != '.'
-  then
-    s ^ "."
-  else
-    s
-
-let rtrim = String.rtrim
-
-let vgap = "\n\\vspace{1cm}\n"
-let hgap = "\\hspace{0.5cm}"
-
 let is_prim_type = function
   | String | Int | Float | Bool | DateTime -> true
   | _ -> false
 
 let is_prim_opt_type = function
   | None -> true
-  | (Some (ty,_)) -> is_prim_type ty
+  | Some (ty,_) -> is_prim_type ty
 
 let rec of_ty_verbatim = function
-  | String -> "string" | Int -> "int" | Float -> "float" | Bool -> "bool"
-  | DateTime -> "datetime" | Enum (name, things) -> name
-  | Set x -> (of_ty_verbatim x) ^ " set"
-  | Map (a, b) -> "(" ^ (of_ty_verbatim a) ^ " -> " ^ (of_ty_verbatim b) ^ ") map"
+  | String -> "string"
+  | Int -> "int"
+  | Float -> "float"
+  | Bool -> "bool"
+  | DateTime -> "datetime"
+  | Enum (name, things) -> name
+  | Set x -> sprintf "%s set" (of_ty_verbatim x)
+  | Map (a, b) -> sprintf "(%s -> %s) map" (of_ty_verbatim a) (of_ty_verbatim b)
   | Ref obj -> obj ^ " ref"
   | Record obj -> obj ^ " record"
 
 
 let rec of_ty = function
-  | String -> "string" | Int -> "int" | Float -> "float" | Bool -> "bool"
-  | DateTime -> "datetime" | Enum (name, things) -> escape name
+  | String -> "string"
+  | Int -> "int"
+  | Float -> "float"
+  | Bool -> "bool"
+  | DateTime -> "datetime"
+  | Enum (name, things) -> escape name
   | Set x -> (of_ty x) ^ " set"
-  | Map (a, b) -> "(" ^ (of_ty a) ^ " $\\rightarrow$ " ^ (of_ty b) ^ ") map"
-  | Ref obj -> (escape obj)^" ref"
-  | Record obj -> escape obj ^ " record"
+  | Map (a, b) -> "(" ^ (of_ty a) ^ " &rarr; " ^ (of_ty b) ^ ") map"
+  | Ref obj -> obj^" ref"
```

Comment:
Do we also need an escape here for `Ref` and `Record`, as class names can contain the `_` character?